pkg/pfcpsim/pfcpsim-client/pfcpsim-client.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Open Networking Foundation
+
+package pfcpsim_client
+
+import (
+	"github.com/c-robinson/iplib"
+	"github.com/omec-project/pfcpsim/pkg/pfcpsim"
+	log "github.com/sirupsen/logrus"
+	"github.com/wmnsk/go-pfcp/ie"
+	"net"
+	"time"
+)
+
+const (
+	ActionForward uint8 = 0x2
+	ActionDrop    uint8 = 0x1
+	ActionBuffer  uint8 = 0x4
+	ActionNotify  uint8 = 0x8
+)
+
+type MockSMF struct {
+	ueAddressPool string
+	nodeBAddress  string
+	upfAddress    string
+
+	activeSessions int
+
+	lastUEAddress net.IP
+
+	client *pfcpsim.PFCPClient
+}
+
+func NewMockSMF(lAddr string, ueAddressPool string, nodeBAddress string, upfAddress string) *MockSMF {
+
+	pfcpClient := pfcpsim.NewPFCPClient(lAddr)
+
+	return &MockSMF{
+		ueAddressPool: ueAddressPool,
+		nodeBAddress:  nodeBAddress,
+		upfAddress:    upfAddress,
+		client:        pfcpClient,
+	}
+}
+
+func (m *MockSMF) Disconnect() {
+	m.client.DisconnectN4()
+	log.Infof("PFCP client Disconnected")
+}
+
+func (m *MockSMF) Connect(remoteAddress string) error {
+	err := m.client.ConnectN4(remoteAddress)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("PFCP client is connected")
+	return nil
+}
+
+// TeardownAssociation uses the PFCP client to tearing down an already established association.
+// If called while no association is established by PFCP client, the latter will return an error
+func (m *MockSMF) TeardownAssociation() {
+	err := m.client.TeardownAssociation()
+	if err != nil {
+		log.Errorf("Error while tearing down association: %v", err)
+		return
+	}
+
+	log.Infoln("Teardown association completed")
+}
+
+// SetupAssociation uses the PFCP client to establish an association,
+// logging its success by checking PFCPclient.IsAssociationAlive
+func (m *MockSMF) SetupAssociation() {
+	err := m.client.SetupAssociation()
+	if err != nil {
+		log.Errorf("Error while setting up association: %v", err)
+		return
+	}
+
+	time.Sleep(pfcpsim.DefaultHeartbeatPeriod)
```

Comment:
I don't think we need to wait here. We do this in integration tests, but this is just for testing purposes, to verify that a connection is maintained by peers.